core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/keystore/persistence/KeyStoreRegistryPersistenceManager.java

```diff
@@ -0,0 +1,421 @@
+                Resource pubKeyResource = registry.get(associations[0].getDestinationPath());
+                keyStoreMetadata.setPublicCertId(pubKeyResource.getProperty(PROP_TENANT_PUB_KEY_FILE_NAME_APPENDER));
+                keyStoreMetadata.setPublicCert((byte[]) pubKeyResource.getContent());
+            }
+        }
+        return keyStoreMetadata;
+    }
+
+    /**
+     * Get the last modified date of the key store.
+     *
+     * @param keyStoreName Key store name.
+     * @param tenantId     Tenant Id.
+     * @return Last modified date of the key store.
+     */
+    public Date getKeyStoreLastModifiedDate(String keyStoreName, int tenantId) {
+
+        Registry registry = getGovernanceRegistry(tenantId);
+        String path = RegistryResources.SecurityManagement.KEY_STORES + REGISTRY_PATH_SEPARATOR + keyStoreName;
+        try {
+            return registry.get(path).getLastModified();
+        } catch (RegistryException e) {
+            LOG.error("Error while getting the last modified date of the key store: " + keyStoreName, e);
+            return null;
+        }
+    }
+
+    /**
+     * Update the key store in the registry.
+     *
+     * @param keyStoreName Key store name.
+     * @param keyStore     Updated key store.
+     * @param tenantId     Tenant Id.
+     */
+    public void updateKeyStore(String keyStoreName, KeyStore keyStore, int tenantId) {
+
+        Registry registry = getGovernanceRegistry(tenantId);
+        String path = RegistryResources.SecurityManagement.KEY_STORES + REGISTRY_PATH_SEPARATOR + keyStoreName;
+        try (ByteArrayOutputStream outputStream = new ByteArrayOutputStream()) {
+            Resource resource = registry.get(path);
+            char[] passwordChar = getKeyStorePasswordFromRegistryResource(resource, keyStoreName);
+            keyStore.store(outputStream, passwordChar);
+            outputStream.flush();
+            resource.setContent(outputStream.toByteArray());
+            registry.put(path, resource);
+            resource.discard();
+        } catch (IOException | KeyStoreException | NoSuchAlgorithmException | CertificateException |
+                 RegistryException e) {
+            throw new SecurityException("Error updating tenanted key store: " + keyStoreName, e);
+        }
+    }
+
+    /**
+     * Delete the key store from the registry.
+     *
+     * @param keyStoreName Name of the key store.
+     * @param tenantId     Tenant Id.
+     * @throws SecurityException If an error occurs while deleting the key store.
+     */
+    public void deleteKeyStore(String keyStoreName, int tenantId) throws SecurityException {
+
+        Registry registry = getGovernanceRegistry(tenantId);
+        String path = RegistryResources.SecurityManagement.KEY_STORES + REGISTRY_PATH_SEPARATOR + keyStoreName;
+        try {
+            org.wso2.carbon.registry.api.Association[] allAssociations = registry.getAllAssociations(path);
+            if (allAssociations.length > 0) {
```

Comment:
Can we be sure `allAssociations` not gonna be null?

Reply:
Addressed with https://github.com/wso2/carbon-kernel/compare/63f3d33d1ed8704022f55645be6b047a005641fb..6c4a7e5fc5efca69a114e5d3a08289b8dc809be7